### PR TITLE
Align contributing guidelines

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: landing-page/content/common/contribute.md [hunk 1]
```diff
@@ -83,6 +83,69 @@ Follow the instructions for [Eclipse](https://github.com/google/google-java-form
...
+for developers and users of the iceberg libraries as APIs and implementations evolve.  The requirements and
+guarantees provided depend on the subproject as described below:
+
+### Major Version Deprecations Required
+
+__Modules__
+`iceberg-api`
+
+The API subproject is the main interface for developers and users of the Iceberg API and therefore has the strongest
+guarantees.  Evolution of the interfaces in this subproject are enforced by [Revapi](https://revapi.org/) and require
```
Comment (jackye1995) [inline landing-page/content/common/contribute.md:98]: nit: extra space here `guarantees.  Evolution`

Can we just use newline for each sentence? That helps making future changes easier to minimize line changes.

File: landing-page/content/common/contribute.md [hunk 2]
```diff
@@ -198,6 +261,87 @@ When passing boolean arguments to existing or external methods, use inline comme
...
+    .untilAsserted(() -> assertThat(tables()).isEmpty());
+```
+
+Please refer to the [usage guide](https://github.com/awaitility/awaitility/wiki/Usage) of [Awaitility](https://github.com/awaitility/awaitility) for more usage examples.
+
+
+### JUnit4 / JUnit5
+
+Iceberg currently uses a mix of JUnit4 and JUnit5 tests. To allow an easier migration to JUnit5 in the future, new test classes
+that are being added to the codebase should be written purely in JUnit5 where possible.
```
Comment (jackye1995) [inline landing-page/content/common/contribute.md:342]: can we give some examples of what is junit4 vs 5? I think people unintentionally use the wrong version sometimes.

File: landing-page/content/common/contribute.md [hunk 3]
```diff
@@ -198,6 +261,87 @@ When passing boolean arguments to existing or external methods, use inline comme
...
+with a default polling interval of **100 millis**:
+
+```java
+deleteTablesAsync();
+Awaitility.await("Tables were not deleted")
+    .atMost(5, TimeUnit.SECONDS)
+    .untilAsserted(() -> assertThat(tables()).isEmpty());
+```
+
+Please refer to the [usage guide](https://github.com/awaitility/awaitility/wiki/Usage) of [Awaitility](https://github.com/awaitility/awaitility) for more usage examples.
```
Comment (jackye1995) [inline landing-page/content/common/contribute.md:336]: we should probably also create an GitHub issue tracking deprecation of the usage of thread sleep, like what we are doing for assert helpers.

File: landing-page/content/common/contribute.md (hunk 1 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:98]: good point, done

File: landing-page/content/common/contribute.md (hunk 3 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:338]: makes sense, I'll open one

File: landing-page/content/common/contribute.md (hunk 2 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:344]: I've adjusted the sentence to:
> Iceberg currently uses a mix of JUnit4 (`org.junit` imports) and JUnit5 (`org.junit.jupiter.api` imports) tests.

This should hopefully make it clearer which imports are being used for which JUnit version

**Review by jackye1995 — APPROVED** [review]

File: landing-page/content/common/contribute.md (hunk 3 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:338]: opened https://github.com/apache/iceberg/issues/7154

File: landing-page/content/common/contribute.md (hunk 2 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:344]: @jackye1995 do you mean an issue to convert JUnit4 tests to JUnit5?

**Comment by jackye1995** [comment]
I will go ahead to merge it since it is already discussed in community sync, we can update wording in subsequent PRs if necessary. Thanks @nastra !

File: landing-page/content/common/contribute.md (hunk 2 above)
Comment (nastra) [reply landing-page/content/common/contribute.md:344]: I've opened https://github.com/apache/iceberg/issues/7160

**Pull request merged**